### DISCUSSION TITLE
OSX alternate-browser script

### DISCUSSION THREAD

**Discussion by ghost** [PDConSec/vsc-print#223] [Q&A] [answered]
I'm on macOS Ventura, default browser is Safari. 
Trying to set up Chrome as the alternate browser. 

- Followed the advice to use a shell script to launch Chrome. 
- But what I have so far launches Chrome (blank) but does not render the file. 
- I have a shell script `launch-chrome.sh`
- It's content is: `open -a "Google Chrome"` 
- Browser Path setting is `~/launch-chrome.sh`
- So it launches Chrome, but blank, no rendered document.
- I assume I'm missing something in my shell script

**Comment by PeterWone** [comment]
Hi Joe. 

I've been busy with working and also building a cabin in the woods so I haven't been keeping up with things. Thank you for the detailed report, I have a Mac now so I'll test this. Does the page render in Safari?

I was going to say check permissions on the bash script but if Chrome is launching it won't be that.

A URL parameter is passed to the script. You don't say whether you are passing this through to Chrome.

**Comment by DavidRawling** [comment]
@joeballard It sounds like you'll need to update your script to pass the parameters to Chrome. In this case, you'll likely want the `$@` placeholder as this contains all parameters other than the script name. That might look something like this:

```
open -a "Google Chrome" $@
```

I also notice though that you're using open as your command. On my system that's an alias for xdg-open, which doesn't accept the `-a` parameter you've got, so I can't be sure that this is exactly what you need.

**Comment by PeterWone** [comment]
I did some research based on everything said so far.

* OSX bash supports `$@`
* `open -a "application name"` is the correct way to do this for OSX

Edit your file to contain 

```bash
open -a "Google Chrome" $@
```

That should work. Please let me know if there are any problems.

**Comment by ghost** [comment]
Adding the '$@' worked fine. Thanks!! And good luck with the cabin in the woods.

**Comment by PeterWone** [comment, marked as answer]
```bash
open -a "Google Chrome" $@
```

Let's break this down.

* On OSX, `open -a "application name"` runs an application by name, which is path independent. In this case the application is "Google Chrome". The double-quotes are required when there is a space in the name.
* `$@` means pass the entire command line apart from the application name. It differs from `$*` in that `$*` parses the parameters first and passes them individually, which can mess with whitespace and order. `$@` passes the whole thing through as one big string.
